charts/linkerd2/templates/web.yaml

```diff
@@ -49,6 +67,11 @@ spec:
       {{.Values.global.controllerComponentLabel}}: web
       {{.Values.global.controllerNamespaceLabel}}: {{.Values.global.namespace}}
       {{- include "partials.proxy.labels" $tree.Values.global.proxy | nindent 6}}
+  {{- if .Values.enablePodAntiAffinity }}
+  strategy:
+    rollingUpdate:
+      maxUnavailable: 1
+  {{- end }}
```

Comment:
I'm not sure that we want this for the web deployment. In HA mode, we don't scale the web replicas up to 3 like we do with other controller components; we keep it at 1.

This is because unlike the other controller components, it's not necessary for linkerd to continue working properly if the web deployment is unavailable during an update.

Reply:
I agree with @kleimkuhler and think we shouldn't add a PDB to the web deployment.  Essentially, adding a PDB would prevent scaling down the cluster if doing so would de-schedule more than one web pod.  Since web is not a critical component, I think this should not prevent scaling events.

Reply:
I've backed out this change and the addition of a PDB to the tap component as this appears to have been extracted into a separate linkerd-viz chart

Reply:
Great thanks @tustvold! Pending CI this looks good 👍